lib/msal-angular/src/msal.guard.ts

```diff
@@ -76,6 +76,15 @@ export class MsalGuard implements CanActivate {
         }
         this.authService.getLogger().verbose("MSAL Guard activated");
 
+        /*
+         * If a page with MSAL Guard is set as the redirect for acquireTokenSilent,
+         * short-circuit to prevent redirecting or popups.
+         */
+        if (UrlString.hashContainsKnownProperties(window.location.hash) && BrowserUtils.isInIframe()) {
```

Comment:
Can you add a TODO here, as this will need to be updated once #2593 is merged. (as we'll need to allow the guard to run in iframes if `allowRedirectInIframe` is set to `true`).

Reply:
Makes sense. Added.